src/lib/proxy/proxy-service.ts

```diff
@@ -102,8 +102,8 @@ export class ProxyService {
         const oldClient = await this.clientForProxyToken(token);
         const oldDefinitions = oldClient.getFeatureToggleDefinitions() || [];
 
-        const newClient = await this.newClientForProxyToken(token);
-        const newDefinitions = newClient.getFeatureToggleDefinitions() || [];
+        const newDefinitions =
+            this.globalFrontendApiCache.getToggles(token) || [];
```

Comment:
we can get all definitions directly from cache since it already does project and env filtering